src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,199 @@
+
+    if (normalizedTitle) {
+      const newTodo: Todo = {
+        title: normalizedTitle,
+        userId: USER_ID,
+        id: 0,
+        completed: false,
+      };
+
+      setTempTodo(newTodo);
+
+      addTodo(
+        newTodo,
+        () => {
+          setNewTodoTitle('');
+        },
+        () => {
+          setTempTodo(null);
+          setTimeout(() => {
+            newTodoInput.current?.focus();
+          }, 5);
+        },
+      );
+    } else {
+      setErrorMessage('Title should not be empty');
+    }
+  };
+
+  // const handleToggleAllChanged = () => {
+  //   // eslint-disable-next-line no-restricted-syntax
+  //   todos.forEach(todo => {
+  //     dispatch({
+  //       type: TodoAction.Update,
+  //       todo: {
+  //         ...todo,
+  //         completed: !isAllCompleted,
+  //       },
+  //     });
+  //   });
+  // };
+
+  const handleCleanCompletedButtonClicked = () => {
+    const completedTodos = todos.filter(({ completed }) => completed);
+
+    dispatch({
+      type: TodoAction.SetOperatingTodos,
+      todos: completedTodos,
+    });
+
+    clearCompleted(completedTodos);
+  };
+
+  const handleErrorHiding = () => {
+    if (errorMessage) {
+      if (contextErrorMessage) {
+        dispatch({
+          type: TodoAction.SetError,
+          errorMessage: '',
+        });
+      }
+
+      setErrorMessage('');
+    }
+  };
+
+  const uncompletedTodosCount
+    = todos.filter(({ completed }) => !completed).length;
+
+  if (contextErrorMessage && errorMessage !== contextErrorMessage) {
+    setErrorMessage(contextErrorMessage);
+  }
+
+  if (errorMessage) {
+    setTimeout(() => {
+      handleErrorHiding();
+    }, 3000);
+  }
+
+  return (
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          {/* this buttons is active only if there are some active todos */}
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            // onClick={handleToggleAllChanged}
+            type="button"
+            className={cn('todoapp__toggle-all', {
+              active: isAllCompleted,
+            })}
+            data-cy="ToggleAllButton"
+          />
+
+          <form onSubmit={handleSubmit}>
+            <input
+              // eslint-disable-next-line jsx-a11y/no-autofocus
+              autoFocus
+              ref={newTodoInput}
+              disabled={!!tempTodo}
+              value={newTodoTitle}
+              data-cy="NewTodoField"
+              type="text"
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+              onChange={event => setNewTodoTitle(event.target.value)}
+            />
+          </form>
+        </header>
+
+        <TodoList todos={filteredTodos} />
+
+        {tempTodo && (
+          <TodoItem todo={tempTodo} isLoading />
+        )}
```

Comment:
```suggestion
        {!!tempTodo && (
          <TodoItem todo={tempTodo} isLoading />
        )}
```
in rendering with && for best rendering control use !! to return a boolean value